HW-DB

Max-Pochepets opened the pull request:
DATABASE connection established

Max-Pochepets committed:
before deleting commented code

Max-Pochepets committed:
Initial

Max-Pochepets committed:
id autoincrement fixed

Max-Pochepets committed:
Refactor:

repeated logic generalized;
custom exception created

Max-Pochepets committed:
`update()` method improved

Max-Pochepets committed:
checkstyle

vicire commented on pom.xml:
```diff
@@ -18,6 +18,14 @@
         </maven.checkstyle.plugin.configLocation>
     </properties>
 
```
remove empty line here and under

vicire commented on src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:
```diff
@@ -0,0 +1,145 @@
+            PreparedStatement preparedStatement
+                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            setUpdate(preparedStatement, manufacturer).executeUpdate();
+
```
remove empty lines :)

vicire commented on src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:
```diff
@@ -0,0 +1,145 @@
+
+            return manufacturer;
+        } catch (SQLException throwables) {
+            throw new RuntimeException("Couldn't establish connection to MySQL server. "
```
look in common mistakes. You need to create the custom exception.

vicire commented on src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:
```diff
@@ -0,0 +1,145 @@
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery(query);
+            Manufacturer manufacturer = new Manufacturer("", "");
+            while (resultSet.next()) {
```
Same here - you can create a private method for logic in cycle

vicire commented on src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:
```diff
@@ -0,0 +1,145 @@
+                    = setUpdate(connection.prepareStatement(query), manufacturer);
+            preparedStatement.setLong(3, manufacturer.getId());
+            preparedStatement.executeUpdate();
+            preparedStatement.close();
```
Honestly, that`s enough for this method) you can return manufacturer, that you already have

vicire commented on src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:
```diff
@@ -0,0 +1,145 @@
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
```
Create constants for magic

Max-Pochepets replied:
I can't, cause this indexes don't serve the same purpose every time they are used.

vicire reviewed (changes requested):
Nice)

Max-Pochepets committed:
Refactor:

got all statements to try with resources closure
added objects that failed to proceed to exception message
deleted unused query
changed util methods to non-static

Sofasmile commented on src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:
```diff
@@ -0,0 +1,121 @@
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers (name,country) VALUES (?,?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement
+                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
```
```suggestion
        try (Connection connection = ConnectionUtil.getConnection();
            PreparedStatement preparedStatement
                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
```
and then we can remove `statement.close()`

Sofasmile commented on src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:
```diff
@@ -0,0 +1,121 @@
+            preparedStatement.close();
+            return manufacturer;
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Couldn't create manufacturer.",
```
which manufacturer can not be created?

Sofasmile commented on src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:
```diff
@@ -0,0 +1,121 @@
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE manufacturers SET name = ?, country = ?"
+                + " WHERE id = ? AND deleted = FALSE";
+        String getNewQuery = "SELECT * FROM manufacturers WHERE id = " + manufacturer.getId();
```
do we need this query here?

Sofasmile commented on src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:
```diff
@@ -0,0 +1,121 @@
+            statement.close();
+            return Optional.ofNullable(manufacturer);
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Couldn't get manufacturer by id.",
```
let's make all messages more informative

Sofasmile commented on src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:
```diff
@@ -0,0 +1,121 @@
+        }
+    }
+
+    private static Manufacturer setManufacturer(ResultSet resultSet) throws SQLException {
```
why is this method static?
